chain/x/zoracle/exec_env.go

```diff
@@ -63,20 +116,19 @@ func (env *ExecutionEnvironment) GetValidatorAddress(validatorIndex int64) ([]by
 	return env.request.RequestedValidators[validatorIndex], nil
 }
 
-func (env *ExecutionEnvironment) GetMaximumResultSize() int64 {
-	return int64(env.keeper.GetParam(env.ctx, KeyMaxResultSize))
-}
-
-func (env *ExecutionEnvironment) GetMaximumCalldataOfDataSourceSize() int64 {
-	return int64(env.keeper.GetParam(env.ctx, KeyMaxCalldataSize))
-}
-
 func (env *ExecutionEnvironment) RequestExternalData(
 	dataSourceID int64,
 	externalDataID int64,
 	calldata []byte,
-) error {
-	return env.keeper.AddNewRawDataRequest(env.ctx, env.requestID, types.ExternalID(externalDataID), types.DataSourceID(dataSourceID), calldata)
+) {
+	// when size of env.rawDataRequests == env.maxRawDataRequest that mean it exceeds the number of request
```

Comment:
// We can safely skip appending rawDataRequests if its size already surpasses maxRawDataRequestCount threshold since it will fail the transaction regardless.